inference-engine/src/legacy_api/src/cnn_network_impl.cpp

```diff
@@ -78,6 +86,23 @@ ICNNNetwork::~ICNNNetwork() {}
 
 CNNNetworkImpl::CNNNetworkImpl() {}
 
+CNNNetworkImpl::CNNNetworkImpl(const ICNNNetwork & ngraphImpl,
+    const ngraph::pass::PassParam::param_callback & callback,
+    bool keep_constant_inputs) {
+    auto ngraphImplPtr = dynamic_cast<const details::CNNNetworkNGraphImpl*>(&ngraphImpl);
+    IE_ASSERT(ngraphImplPtr != nullptr);
+    IE_ASSERT(ngraphImplPtr->getFunction() != nullptr);
+    auto graph = ngraphImplPtr->cloneFunction();
+    // Disable shape inference (WA for generic operations)
+    ::ngraph::op::GenericIE::DisableReshape noReshape(graph);
+
+    ::ngraph::pass::CommonOptimizations(callback).run_on_function(graph);
```

Comment:
Plugins may have different conversion logic. For example in VPU we are going to add transformation after Conversion to legacy so it becomes impossible with this approach.

Reply:
Old approach is still available. This one is just a default conversion + callback.

Reply:
BTW, as discussed with @gladilov-gleb VPU still have manual pipeline in their code and all transformations can be inserted up to them. The same for KMB plugin

Reply:
Is there a chance operations that moved to ctor (default way to convert) could be called twice?

Reply:
discussed with @gladilov-gleb offline. Conversion cannot be called 2 times because `CNNNetworkImpl` is got using either this new constructor or manual pipeline